contracts/TokenDistributor.sol

```diff
@@ -118,42 +120,48 @@ contract TokenDistributor is ITokenDistributor {
     /// @inheritdoc ITokenDistributor
     function claimEpoch(address to, uint epochNumber) public override {
         _checkpointWhenReady();
-        _claimEpoch(to, epochNumber);
+        uint256 proportionalFees = _claimEpoch(to, epochNumber);
+
+        lastTokenBalance -= proportionalFees;
+
+        rewardsToken.transfer(to, proportionalFees);
     }
 
     /// @notice internal claimEpoch function
-    function _claimEpoch(address to, uint epochNumber) internal {
+    function _claimEpoch(address to, uint epochNumber) internal returns (uint256 proportionalFees) {
         _isEpochReady(epochNumber);
-        mapping(uint256 => bool) storage claimedEpochsTo = claimedEpochs[to];
-        if (claimedEpochsTo[epochNumber]) {
+        if (_claimedEpochsBitMap[to].get(epochNumber)) {
```

Comment:
Can you not use your `claimedEpochs()` view function here?

Reply:
I think you meant to ask me to use `claimedEpochs()` function here. It's slightly less gas efficient to use it, but it's more dev friendly. I will make the change.

Reply:
Done.

Reply:
Yeah I think you can also follow the `_claimedEpochs()` internal `claimedEpochs()` external format if you really want the gas savings. But this is probably good enough